circom/tests/constraints/with_call_vec_arg.circom

```diff
@@ -25,19 +25,24 @@ template ComputeFee() {
 component main = ComputeFee();
 
 //CHECK-LABEL: define{{.*}} void @..generated..loop.body.
-//CHECK-SAME: [[$F_ID_1:[0-9]+]]([0 x i256]* %lvars, [0 x i256]* %signals, i256* %sig_0, i256* %var_1){{.*}} {
+//CHECK-SAME: [[$F_ID_1:[0-9]+]]([0 x i256]* %lvars, [0 x i256]* %signals, i256* %sig_0){{.*}} {
 //CHECK-NEXT: ..generated..loop.body.[[$F_ID_1]]:
 //CHECK-NEXT:   br label %call1
 //CHECK-EMPTY: 
 //CHECK-NEXT: call1:
 //CHECK-NEXT:   %[[FUN_NAME:[0-9a-zA-Z_.]+]]_arena = alloca [3 x i256], align 8
 //CHECK-NEXT:   %[[T00:[0-9a-zA-Z_.]+]] = getelementptr [3 x i256], [3 x i256]* %[[FUN_NAME]]_arena, i32 0, i32 0
-//CHECK-NEXT:   %[[T01:[0-9a-zA-Z_.]+]] = getelementptr i256, i256* %var_1, i32 0
-//CHECK-NEXT:   %[[COPY_SRC_0:[0-9a-zA-Z_.]+]] = getelementptr i256, i256* %[[T01]], i32 0
+//CHECK-NEXT:   %[[T01:[0-9a-zA-Z_.]+]] = getelementptr [0 x i256], [0 x i256]* %lvars, i32 0, i32 8
+//CHECK-NEXT:   %[[T12:[0-9a-zA-Z_.]+]] = load i256, i256* %[[T01]], align 4
+//CHECK-NEXT:   %[[T13:[0-9a-zA-Z_.]+]] = call i32 @fr_cast_to_addr(i256 %[[T12]])
+//CHECK-NEXT:   %[[T14:[0-9a-zA-Z_.]+]] = mul i32 2, %[[T13]]
+//CHECK-NEXT:   %[[T15:[0-9a-zA-Z_.]+]] = add i32 %[[T14]], 0
+//CHECK-NEXT:   %[[T16:[0-9a-zA-Z_.]+]] = getelementptr [0 x i256], [0 x i256]* %lvars, i32 0, i32 %[[T15]]
+//CHECK-NEXT:   %[[COPY_SRC_0:[0-9a-zA-Z_.]+]] = getelementptr i256, i256* %[[T16]], i32 0
```

Comment:
I realize this looks like a regression; what was fixed indexing via a parameter has become dynamic indexing. However, this is actually caused by a different long-standing issue [https://veridise.atlassian.net/browse/VAN-671](https://veridise.atlassian.net/browse/VAN-671) and the fix to the interpreter has brought it to the surface.